libs/s25main/mapGenerator/Harbors.cpp

```diff
@@ -86,19 +78,29 @@ namespace rttr { namespace mapGenerator {
             {
                 if(isCoast(pt))
                 {
-                    auto coast = Collect(map.textures, pt, isCoast);
+                    const auto coast = Collect(map.textures, pt, isCoast);
                     coasts.push_back(coast);
                     visited.insert(coast.begin(), coast.end());
                 }
             }
         }
+
+        const auto distanceToRiver = DistancesTo(join(rivers), map.size);
+        const auto closeToRiver = [&distanceToRiver](const MapPoint& pt) { return distanceToRiver[pt] < 5; };
+
+        for(std::vector<MapPoint>& coast : coasts)
+        {
+            coast.erase(std::remove_if(coast.begin(), coast.end(), closeToRiver), coast.end());
```

Comment:
Ah, I thought it was somewhere, just didn't remember where to find it! Cheers!

Reply:
containerUtils.h is the header, so for similar stuff look there. I'm open for adding new, commonly used operations which are...unwieldy there or at least "rangify" existing ones (i.e. accept a container instead of iterators)

BTW: I noticed that C++20 introduced `std::erase_if` as a free function, so I guess we should update the names in containerUtils to match that and avoid confusion. Just FYI, I'll likely do that later sometime